src/SonarScanner.MSBuild.Shim/ScannerEngineInput.cs

```diff
@@ -43,11 +43,21 @@ public ScannerEngineInput(AnalysisConfig config)
         };
         modules = new JProperty("value", string.Empty);
         Add("sonar.modules", modules);
+        if (provider.TryGetValue(SonarProperties.SonarToken, out var token))
```

Comment:
This looks all wrong. `ScannerEngineInput` should not be tangled with this sensitive data logic, and should not receive `IAnalysisPropertyProvider` in its constructor.

The CLI does this:
```
var stringArgs = userCmdLineArguments.GetAllProperties().Select(x => x.AsSonarScannerArg()).ToArray();
```
and we should do the same thing. And that logic should remain only in `ScannerEngineInputGenerator`